components/layout/index.jsx

```diff
@@ -1,20 +1,36 @@
-import React from 'react'
+import React, { useState } from 'react'
+import classNames from 'classnames'
 import Header from '../header'
 import Head from '../head'
 import './index.css'
 import Sidebar from '../sidebar'
 
 const Layout = ({ children }) => {
+  const [isSidebarVisible, toggleSidebarVisibility] = useState(false)
+
+  const handleCLick = e => {
+    const { name, value } = e.target
+    if (['sidebar.show', 'sidebar.close'].includes(`${name}.${value}`))
+      toggleSidebarVisibility(!isSidebarVisible)
+  }
+
+  // TODO: Ask Viktor how aria should be handled
+  /* eslint-disable */
   return (
     <>
       <Head />
-      <div className="layout">
+      <div className="layout" onClick={handleCLick}>
         <Header className="layout-bar fixed" />
```

Comment:
I would rather add listener to `Header` to make it as close as possible to the toggle button but not passing it directly.

React does event delegation on mounting point so here it's just matter of taste.

Also, you have a typo in `CLick`  and would rather prefer having more concrete name, i.e. `handleSidebarToggle`, unless you do really complex stuff in the event listener.